libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -645,10 +646,18 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      * navigation simulation, or before resetting location to not real (simulated)
      * position without recreation of [MapboxNavigation].
      */
-    @Deprecated(message = "use a function withe the callback instead")
+    @Deprecated(message = "use the overload with the callback instead")
     fun resetTripSession() {
-        resetTripSession {
-            // no-op
+        // using a blocking function to keep parity with the original implementation so that
+        // Nav Native is fully done with the reset when this function returns
+        runBlocking {
```

Comment:
@LukasPaczos , from the first glance it seems like it should always cause deadlock:

1. User calls `MapboxNavigation#resetTripSession()` from a main thread
2. `runBlocking` stops and holds main thread until it receives callback from NN
3. NN does some heavy actions and posts callback to main thread via events queue 
4. Event's queue aren't processes because main thread has stopped during step 2

I was surprised that it works and it turned out that  NN calls the callback from their worker thread without rescheduling to the main thread. Do you think it's safe to rely on this implementation detail of NN? I worry that if they change it, deprecated version of `resetTripSession` will always cause a deadlock.

Reply:
Thanks for raising this. In the end no, we shouldn't really rely on this. This is a strong enough reason to try and find a different approach or work with Nav Native to leave the old function be, even if deprecated.

Tracking in NAVAND-984.

Reply:
Adding a test as a safeguard that would block us from adopting a NN version that would break this logic - https://github.com/mapbox/mapbox-navigation-android/pull/6710.